compass/ocean/tests/global_ocean/mesh/so12to30/so12to30.cfg

```diff
@@ -2,28 +2,47 @@
 [vertical_grid]
 
 # the type of vertical grid
-grid_type = index_tanh_dz
+grid_type = tanh_dz
 
 # Number of vertical levels
-vert_levels = 64
+vert_levels = 80
 
 # Depth of the bottom of the ocean
 bottom_depth = 5500.0
 
 # The minimum layer thickness
-min_layer_thickness = 10.0
+min_layer_thickness = 2.0
 
 # The maximum layer thickness
-max_layer_thickness = 250.0
+max_layer_thickness = 150.0
 
-# The characteristic number of levels over which the transition between
-# the min and max occurs
-transition_levels = 28
+# Options relate to adjusting the sea-surface height or land-ice pressure
+# below ice shelves to they are dynamically consistent with one another
+[ssh_adjustment]
+
+# the number of iterations of ssh adjustment to perform
+iterations = 4
 
 
 # options for global ocean testcases
 [global_ocean]
 
+## config options related to the initial_state step
+
+# minimum number of vertical levels, both in the open ocean and in ice-shelf
+# cavities
+min_levels = 5
+cavity_min_levels = ${min_levels}
+
+# minimum thickness of layers in ice-shelf cavities at the beginning and end
+# of iterative ssh init
+cavity_min_layer_thickness_initial = 2.0
+cavity_min_layer_thickness_final = 1.0
```

Comment:
@xylar Assuming that we still have a max sea ice pressure-related depression of 5m, I just want to make sure that minimum column thickness > 5m where sea ice is allowed to form. Can you confirm this?

Reply:
@xylar Is this the right file to be looking at? `/lcrc/group/e3sm/ac.xylar/compass_1.4/chrysalis/e3smv3-meshes/sowisc12to30e3r3/ocean/global_ocean/SOwISC12to30/WOA23/files_for_e3sm/ocean_initial_condition/mpaso.SOwISC12to30E3r3.20240705.nc`

Reply:
I think the answer is yes but I would treat this as the "official" one for now:
```
/lcrc/group/e3sm/data/inputdata/ocn/mpas-o/SOwISC12to30E3r3/mpaso.SOwISC12to30E3r3.20240705.nc
```
I believe those 2 files are identical.

Reply:
There is 1 cell outside ice shelf cavities (`landIceMask == 0`) with column thickness < 5.0m (4.4m).
There are 9 cells inside ice shelf cavities (`landIceMask == 1`) with column thickness < 5.0m (min column thickness of 2.1m, 5 active layers with min column thickness of 0.36m).

We definitely need to set the minimum thickness outside cavities with enough of a buffer that it is >5m, since we can apply ~5m worth of sea ice pressure. I'd propose a min column thickness of 10m everywhere. We don't strictly need this in ice shelf cavities since sea ice can't form there but it does seem that we're getting greater ssh perturbations there than elsewhere.

Reply:
@cbegeman, I agree.  Let's stick with a 10-m minimum water-column thickness both inside and outside the cavities.

I'm adding a histogram (and a min/max bit of text) to the initial-condition plots to at least partly cover this sanity check.

Reply:
@cbegeman, I synced over the new version of the file.  Could you give it another look and make sure you see in the E3SM initial condition what I was seeing in the MPAS-Ocean initial condition, namely that the minimum column thickness isn't too much thinner than 10 m?

```
/lcrc/group/e3sm/data/inputdata/ocn/mpas-o/SOwISC12to30E3r3/mpaso.SOwISC12to30E3r3.20240711.nc
```
(Same as before but with today's date.)

Reply:
@xylar This is looking good to me! Thanks

Number of cells with column thickness less than 10m inside ice shelf cavities	61 (min column thickness 5.5m)
Number of cells with column thickness less than 10m outside ice shelf cavities 	10 (min column thickness 8.3m)
Number of cells with column thickness less than 5m     0

Reply:
Okay, it's interesting how much some water columns still thin during dynamic adjustment!  I hadn't realized it would be that much.  It's certainly better to have more of a safety margin!

Reply:
At first, I was worried I messed something up with MPAS-Ocean the initial condition (before dynamic adjustment) but that doesn't seem to be the reason there were such thin water columns in the E3SM initial condition.